lib/aws/active_job/sqs/executor.rb

```diff
@@ -57,6 +65,9 @@ def shutdown(timeout = nil)
                          'the queue and can be ru-run once their visibility timeout ' \
                          'passes.'
           end
+          @error_queue.push(nil) # process any remaining errors and then terminate
```

Comment:
Is there any chance this is never reached? Will you need to ensure this?

Reply:
I don't believe so - the executor's `shutdown` and `wait_for_termination` won't raise errors.  Any exceptions that happen during the shutdown are in workers and will be pushed to the error queue.  

Note - there is a case where some exceptions may not be handled after a terminal exception has already been raised, but I think thats intended.